commit bdd2d86749326d9ec7fce75f25efb5e46ace562f
Author: Susan Franklin <sfranklin>
Date:   Thu Apr 23 20:43:46 2009 +0000

    Bug 272795 - [ui] Bad layout and cropped information in install dialogs

12	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/AvailableIUsPage.java
2	2	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ILayoutConstants.java
4	4	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ResolutionResultsWizardPage.java
1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/SelectableIUsPage.java
2	2	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/IUGeneralInfoPropertyPage.java
3	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/RepositorySelectionGroup.java
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/AvailableIUsPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/AvailableIUsPage.java
index bb4c951..ad8b4d9 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/AvailableIUsPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/AvailableIUsPage.java
@@ -30,10 +30,10 @@ import org.eclipse.jface.dialogs.IDialogSettings;
 import org.eclipse.jface.resource.JFaceResources;
 import org.eclipse.jface.viewers.*;
 import org.eclipse.swt.SWT;
+import org.eclipse.swt.custom.SashForm;
 import org.eclipse.swt.dnd.*;
 import org.eclipse.swt.events.*;
-import org.eclipse.swt.layout.GridData;
-import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.layout.*;
 import org.eclipse.swt.widgets.*;
 
 public class AvailableIUsPage extends ProvisioningWizardPage implements ISelectableIUsPage {
@@ -92,12 +92,18 @@ public class AvailableIUsPage extends ProvisioningWizardPage implements ISelecta
 		// Repo manipulation 
 		createRepoArea(composite);
 
+		SashForm sashForm = new SashForm(composite, SWT.VERTICAL);
+		FillLayout fill = new FillLayout();
+		sashForm.setLayout(fill);
+		GridData data = new GridData(GridData.FILL_BOTH);
+		sashForm.setLayoutData(data);
+
 		// Now the available group 
 		// If we have a repository manipulator, we want to default to showing no repos.  Otherwise all.
 		int filterConstant = AvailableIUGroup.AVAILABLE_NONE;
 		if (policy.getRepositoryManipulator() == null)
 			filterConstant = AvailableIUGroup.AVAILABLE_ALL;
-		availableIUGroup = new AvailableIUGroup(policy, composite, JFaceResources.getDialogFont(), manager, queryContext, ProvUI.getIUColumnConfig(), filterConstant);
+		availableIUGroup = new AvailableIUGroup(policy, sashForm, JFaceResources.getDialogFont(), manager, queryContext, ProvUI.getIUColumnConfig(), filterConstant);
 
 		// Selection listeners must be registered on both the normal selection
 		// events and the check mark events.  Must be done after buttons 
@@ -122,9 +128,11 @@ public class AvailableIUsPage extends ProvisioningWizardPage implements ISelecta
 		activateCopy(availableIUGroup.getStructuredViewer().getControl());
 
 		// Details area
-		iuDetailsGroup = new IUDetailsGroup(composite, availableIUGroup.getStructuredViewer(), convertHorizontalDLUsToPixels(DEFAULT_WIDTH), false);
+		iuDetailsGroup = new IUDetailsGroup(sashForm, availableIUGroup.getStructuredViewer(), convertHorizontalDLUsToPixels(DEFAULT_WIDTH), true);
 		detailsArea = iuDetailsGroup.getDetailsArea();
 
+		sashForm.setWeights(ILayoutConstants.IUS_TO_DETAILS_WEIGHTS);
+
 		// Controls for filtering/presentation/site selection
 		Composite controlsComposite = new Composite(composite, SWT.NONE);
 		layout = new GridLayout();
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ILayoutConstants.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ILayoutConstants.java
index 30cabb8..de7a2b5 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ILayoutConstants.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ILayoutConstants.java
@@ -20,11 +20,11 @@ package org.eclipse.equinox.internal.p2.ui.dialogs;
  * @since 3.5
  */
 public interface ILayoutConstants {
-	public static final int DEFAULT_DESCRIPTION_HEIGHT = 3;
+	public static final int DEFAULT_DESCRIPTION_HEIGHT = 4;
 	public static final int DEFAULT_SITEDETAILS_HEIGHT = 2;
 	public static final int DEFAULT_COLUMN_WIDTH = 60;
 	public static final int DEFAULT_SMALL_COLUMN_WIDTH = 20;
 	public static final int DEFAULT_TABLE_HEIGHT = 15;
 	public static final int DEFAULT_TABLE_WIDTH = 120;
-
+	public static final int[] IUS_TO_DETAILS_WEIGHTS = new int[] {80, 20};
 }
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ResolutionResultsWizardPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ResolutionResultsWizardPage.java
index 572c36a..27953e1 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ResolutionResultsWizardPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ResolutionResultsWizardPage.java
@@ -11,18 +11,18 @@
  *******************************************************************************/
 package org.eclipse.equinox.internal.p2.ui.dialogs;
 
-import org.eclipse.equinox.internal.provisional.p2.ui.model.IUElementListRoot;
-
 import org.eclipse.core.runtime.Assert;
 import org.eclipse.core.runtime.IStatus;
 import org.eclipse.equinox.internal.p2.ui.ProvUIMessages;
-import org.eclipse.equinox.internal.p2.ui.model.*;
+import org.eclipse.equinox.internal.p2.ui.model.ElementUtils;
+import org.eclipse.equinox.internal.p2.ui.model.QueriedElement;
 import org.eclipse.equinox.internal.p2.ui.viewers.IUDetailsLabelProvider;
 import org.eclipse.equinox.internal.provisional.p2.director.ProvisioningPlan;
 import org.eclipse.equinox.internal.provisional.p2.engine.ProvisioningContext;
 import org.eclipse.equinox.internal.provisional.p2.metadata.IInstallableUnit;
 import org.eclipse.equinox.internal.provisional.p2.query.IQueryable;
 import org.eclipse.equinox.internal.provisional.p2.ui.ProvisioningOperationRunner;
+import org.eclipse.equinox.internal.provisional.p2.ui.model.IUElementListRoot;
 import org.eclipse.equinox.internal.provisional.p2.ui.operations.PlannerResolutionOperation;
 import org.eclipse.equinox.internal.provisional.p2.ui.operations.ProfileModificationOperation;
 import org.eclipse.equinox.internal.provisional.p2.ui.policy.Policy;
@@ -128,7 +128,7 @@ public abstract class ResolutionResultsWizardPage extends ResolutionStatusPage {
 
 		updateStatus(input, resolvedOperation);
 		setControl(sashForm);
-		sashForm.setWeights(new int[] {80, 20});
+		sashForm.setWeights(ILayoutConstants.IUS_TO_DETAILS_WEIGHTS);
 		Dialog.applyDialogFont(sashForm);
 	}
 
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/SelectableIUsPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/SelectableIUsPage.java
index 2da9dcc..d27a1a2 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/SelectableIUsPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/SelectableIUsPage.java
@@ -134,7 +134,7 @@ public class SelectableIUsPage extends ResolutionStatusPage implements IResoluti
 		detailsArea = iuDetailsGroup.getDetailsArea();
 		updateStatus(root, resolvedOperation);
 		setControl(sashForm);
-		sashForm.setWeights(new int[] {80, 20});
+		sashForm.setWeights(ILayoutConstants.IUS_TO_DETAILS_WEIGHTS);
 		Dialog.applyDialogFont(sashForm);
 	}
 
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/IUGeneralInfoPropertyPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/IUGeneralInfoPropertyPage.java
index cf2a563..1f38230 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/IUGeneralInfoPropertyPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/IUGeneralInfoPropertyPage.java
@@ -71,8 +71,8 @@ public class IUGeneralInfoPropertyPage extends IUPropertyPage {
 			group.setLayout(new GridLayout());
 			group.setLayoutData(new GridData(GridData.FILL_BOTH));
 
-			Text text = new Text(group, SWT.MULTI | SWT.WRAP | SWT.READ_ONLY);
-			GridData gd = new GridData(GridData.FILL_HORIZONTAL | GridData.VERTICAL_ALIGN_BEGINNING);
+			Text text = new Text(group, SWT.MULTI | SWT.WRAP | SWT.READ_ONLY | SWT.V_SCROLL);
+			GridData gd = new GridData(SWT.FILL, SWT.FILL, true, true);
 			gd.widthHint = computeWidthLimit(text, 80);
 			gd.heightHint = 200;
 			text.setEditable(false);
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/RepositorySelectionGroup.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/RepositorySelectionGroup.java
index 8a5ba12..b028e0b 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/RepositorySelectionGroup.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/provisional/p2/ui/dialogs/RepositorySelectionGroup.java
@@ -63,6 +63,8 @@ public class RepositorySelectionGroup {
 	private static final int DEC_MARGIN_WIDTH = 2;
 	private static final String LINKACTION = "linkAction"; //$NON-NLS-1$
 
+	// see https://bugs.eclipse.org/bugs/show_bug.cgi?id=245569
+	private static final int COUNT_VISIBLE_ITEMS = 20;
 	IWizardContainer container;
 	Policy policy;
 	IUViewQueryContext queryContext;
@@ -126,7 +128,7 @@ public class RepositorySelectionGroup {
 		});
 		// Auto complete - install before our own key listeners, so that auto complete gets first shot.
 		repoAutoComplete = new ComboAutoCompleteField(repoCombo);
-
+		repoCombo.setVisibleItemCount(COUNT_VISIBLE_ITEMS);
 		repoCombo.addKeyListener(new KeyAdapter() {
 
 			public void keyPressed(KeyEvent e) {
